Mage.Sets/src/mage/cards/e/ErietteOfTheCharmedApple.java

```diff
@@ -0,0 +1,127 @@
+package mage.cards.e;
+
+import java.util.Objects;
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfEndStepTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.abilities.effects.common.LoseLifeOpponentsEffect;
+import mage.abilities.effects.common.combat.CantAttackYouAllEffect;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.targetpointer.FixedTarget;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+
+/**
+ *
+ * @author Xanderhall
+ */
+public final class ErietteOfTheCharmedApple extends CardImpl {
+
+    public ErietteOfTheCharmedApple(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{W}{B}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WARLOCK);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(4);
+
+        // Each creature that's enchanted by an Aura you control can't attack you or planeswalkers you control.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new ErietteOfTheCharmedAppleEffect()));
```

Comment:
So you do not need to make a custom effect there I don't think.

Something like that should work:
```java
this.addAbility(new SimpleStaticAbility(
        Zone.BATTLEFIELD,
        new CantAttackYouAllEffect(Duration.WhileOnBattlefield, filter, true)
))
```

You only need to make a filter that correspond to "creature that's enchanted by an Aura you control"

Reply:
[[Elephant Grass]] for reference (the black creature part)

Reply:
That's the issue, I don't quite know how to make that as a filter. I'll have to poke at it more!

Reply:
Oh and while what I said above is correct (if there is an easy way to make a filter, the code above would have worked), a filter would have been a custom variation on `EnchantedPredicate`, so that would not reduce much the custom code compared to a custom Effect.